multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/DetermineServiceKeysToRecreateStep.java

```diff
@@ -102,8 +101,13 @@ private boolean areServiceKeysEqual(CloudServiceKey key1, CloudServiceKey key2)
         return Objects.equals(key1.getCredentials(), key2.getCredentials()) && Objects.equals(key1.getName(), key2.getName());
     }
 
-    private boolean shouldDelete(CloudServiceKey existingKey, List<CloudServiceKey> keys) {
-        return getWithName(keys, existingKey.getName()) == null;
+    public static boolean isMetadataEqual(Metadata metadataA, Metadata metadataB) {
+        if (metadataA == null || metadataB == null) {
+            return metadataA == null && metadataB == null;
+        }
+
+        return Objects.equals(metadataA.getAnnotations(), metadataB.getAnnotations())
+            && Objects.equals(metadataA.getLabels(), metadataB.getLabels());
```

Comment:
Isn't the `Metadata::equals` implemented the same way? 
Why can't we directly use `Objects.equals(metadataA, metadataB)`?